### PR TITLE
Refactor ReleaseFile API and use ULID ids

### PR DESCRIPTION
**Pull request by bloodearnest**
- ignore some 3.9.6 warnings
- Move the Release and ReleaseFile ids to ULIDs
- Refactor release API again

### CODE REVIEW

File: jobserver/api/releases.py [hunk 1]
```diff
@@ -125,21 +119,28 @@ def get(self, request, workspace_name):
...
         File must be listed in Release.requested_files, and the hash must match.
         """
         release = get_object_or_404(Release, id=release_id)
+        backend, user = validate_upload_access(request, release.workspace)
 
-        if filename not in release.requested_files:
-            raise NotFound
+        try:
+            upload = request.data["file"]
+            filename = upload.name
```
Comment (ghickman) [inline jobserver/api/releases.py:136]: Can this line throw a KeyError? if not can we pull it out of the try/catch to leave it focussed?

File: jobserver/releases.py [hunk 2]
```diff
@@ -7,8 +7,29 @@
 from .models.outputs import absolute_file_path
 
 
+class ReleaseFileAlreadyExists(Exception):
+    pass
```
Comment (ghickman) [inline jobserver/releases.py:11]: This makes the API code really readable 👍👍

**Review by ghickman — APPROVED** [review]
Looks great 👍

File: jobserver/api/releases.py (hunk 1 above)
Comment (bloodearnest) [reply jobserver/api/releases.py]: fixed

File: jobserver/releases.py (hunk 2 above)
Comment (bloodearnest) [reply jobserver/releases.py:11]: yeah internal logic exceptions are a Good Idea, I think

**Pull request merged**